sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py

```diff
@@ -569,16 +582,47 @@ def _prepare_to_copy(
             workspace_location,
             workspace_guid,
             AzureMLResourceType.DATA,
-            data.name,
-            data.version,
+            name,
+            version,
         )
 
-        return WorkspaceAssetReference(
-            name=name if name else data.name,
-            version=version if version else data.version,
+        data_ref = WorkspaceAssetReference(
+            name=share_with_name if share_with_name else name,
+            version=share_with_version if share_with_version else version,
             asset_id=asset_id,
         )
 
+        with self._set_registry_client(registry_name):
+            return self.create_or_update(data_ref)
```

Comment:
Should we also update the signature of the create_or_update method to accept WorkspaceAssetReference since data_ref is a WorkspaceAssetReference?

Reply:
WorkspaceAssetReference is actually Asset, so this should be ok..